### PR TITLE
added search funcionality

### PR DESCRIPTION
**Pull request by qbasso**

### CODE REVIEW

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 1]
```diff
@@ -105,6 +106,26 @@ object ConstructorIo {
         return dataManager.getAutocompleteResults(query, params.toTypedArray())
     }
 
+    fun search(text: String, vararg facets: Pair<String, List<String>>, page: Int? = null, perPage: Int? = null, groupId: Int? = null): Observable<ConstructorData<SearchResponse>> {
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt]: Why use `vararg` for facets?

Comment (qbasso) [reply library/src/main/java/io/constructor/core/ConstructorIo.kt]: so you can define define multiple, I believe that it was requirement

Comment (rubikzube) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt]: can we rename to `getSearchResults` to match `getAutocompleteResults`?

Comment (qbasso) [reply library/src/main/java/io/constructor/core/ConstructorIo.kt]: yes

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 2]
```diff
@@ -105,6 +106,26 @@ object ConstructorIo {
...
+        val sessionId = preferenceHelper.getSessionId(sessionIncrementEventHandler)
+        val encodedParams: ArrayList<Pair<String, String>> = arrayListOf()
+        val params: ArrayList<Pair<String, String>> = arrayListOf()
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/core/ConstructorIo.kt]: We create and add items to `params`, but I cannot see where it is read from.

File: library/src/main/java/io/constructor/data/DataManager.kt [hunk 3]
```diff
@@ -25,6 +29,30 @@ constructor(private val constructorApi: ConstructorApi) {
         }
     }.toObservable()
 
+    fun search(text: String, encodedParams: Array<Pair<String, String>> = arrayOf()): Observable<ConstructorData<SearchResponse>> {
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/DataManager.kt]: Can we rename `search` to `getSearchResults` to match `getAutocompleteResults`?

File: library/src/main/java/io/constructor/core/ConstructorIo.kt (hunk 2 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/core/ConstructorIo.kt]: good point, its a bug we should put it to encoded params, as we build serach url dynamically due to shortcomings of http lib we use

**Pull request merged**